docs/user-guide/examples.md

```diff
@@ -130,4 +130,35 @@ defaultEntryPoints = ["http"]
     headerField = "X-WebAuth-User"
     [entryPoints.http.auth.basic]
     users = ["test:$apr1$H6uskkkW$IgXLP6ewTrSuBkTrqE8wj/", "test2:$apr1$d9hr9HBB$4HxwgUir3HP4EsggP/QNo0"]
+```
+
+
+## Enable authentication forwarding
+
+Uses a query param from the request to check for authentication with an auth service.
+If the auth service returns 200 OK the request is forwarded to the backend with additional headers/query params from the auth service.
+The response from the auth service is parsed, json encoded and can be forwarded to the backend as headers/query params.
+The field names in the request and the response can be manipulated for internal use via `as` config for request params 
+and the response from the auth server can be manipulated into a different name via `as` config for the response params.
+```
+defaultEntryPoints = ["http"]
+[entryPoints]
+  [entryPoints.http]
+  address = ":80"
+  [entryPoints.http.auth.forward]
+    address = "http://authserver.com/auth"
+    [entryPoints.http.auth.Forward.requestParameters.email]
+      name = "email"
+      as = "theEmail"
+    [entryPoints.http.auth.Forward.requestParameters.token]
+      name = "token"
+      as = "theToken"
+    [entryPoints.http.auth.Forward.responseReplayFields.userId]
+      path = "user.id"
+      as = "X-User-Id"
+      in = "header"
+    [entryPoints.http.auth.Forward.responseReplayFields.userName]
+      path = "user.name"
+      as = "" # No name transformation
+      in = "parameter"
```

Comment:
I could just add this sequence diagram if this would help clarify the flow.
![traefik authentication forwarding](https://cloud.githubusercontent.com/assets/1196390/22011065/9f4d893e-dc95-11e6-99a9-83fd96ac58f0.png)